src/python_testing/TC_OCC_3_2.py

```diff
@@ -0,0 +1,241 @@
+        
+        self.step("3c")
+        self.wait_for_user_input(prompt_msg="Type any letter and press ENTER after the sensor occupancy is triggered and its occupancy state changed.")
+        
+        self.step("3d")
+        self._await_sequence_of_reports(report_queue=attrib_listener.attribute_queue, endpoint_id=endpoint_id, attribute=cluster.Attributes.Occupancy, sequence=[
+                                        0, 1], timeout_sec=post_prompt_settle_delay_seconds)
+        
+        self.step("4a")
+        if attributes.HoldTime.attribute_id not in attribute_list:
+            logging.info("No HoldTime attribute supports. Terminate this test case")
+            self.skip_all_remaining_steps("4b")
+            
+        self.step("4b")
+        initial_dut = await self.read_occ_attribute_expect_success(endpoint=endpoint, attribute=attributes.HoldTime)
+        
+        self.step("4c")
+        # write a different a HoldTime attibute
+        diff_val = 12
+        await ChipDeviceCtrl.WriteAttribute(node_id, [(endpoint, attributes.HoldTime(diff_val))])
+        
+        self.step("4d")
+        self._await_sequence_of_reports(report_queue=attrib_listener.attribute_queue, endpoint_id=endpoint_id, attribute=cluster.Attributes.HoldTime, sequence=[
+                                        initial_dut, diff_val], timeout_sec=post_prompt_settle_delay_seconds)
+        
+        self.step("5a")
+        if (occupancy_sensor_type_bitmap_dut & Clusters.OccupancySensing.Enums.OccupancySensorTypeEnum.kPir) == 0:
+            logging.info("No PIR timing attribute supports. Skip this test case")
+            self.skip("5b")
+            self.skip("5c")
+            self.skip("5d")
+            
+        self.step("5b")    
+        if attributes.PIROccupiedToUnoccupiedDelay.attribute_id in attribute_list:
+            initial_dut = await self.read_occ_attribute_expect_success(endpoint=endpoint, attribute=attributes.PIROccupiedToUnoccupiedDelay)
+        
+        else:        
+            logging.info("No PIROccupiedToUnoccupiedDelay attribute supports. Terminate this test case")
+        
+        self.step("5c")
+        # write the new attribute value
+        diff_val = 11
+        await ChipDeviceCtrl.WriteAttribute(node_id, [(endpoint, attributes.PIROccupiedToUnoccupiedDelay(diff_val))])
+        
+        self.step("5d")
+        self._await_sequence_of_reports(report_queue=attrib_listener.attribute_queue, endpoint_id=endpoint_id, attribute=cluster.Attributes.PIROccupiedToUnoccupiedDelay, sequence=[
+                                        initial_dut, diff_val], timeout_sec=post_prompt_settle_delay_seconds)
+                                        
+        self.step("6a")
+        if (occupancy_sensor_type_bitmap_dut & Clusters.OccupancySensing.Enums.OccupancySensorTypeEnum.kUltrasonic) == 0:         
+            logging.info("No Ultrasonic timing attribute supports. Skip this test case")
+            self.skip("6b")

```

Comment:
you skip 6b, yet below you calll step 6b ... this seems to be some form of mismatch here

Reply:
if the condition is not met, I wanted to skip the following 3 test steps since those test steps are related together.  So intention is to bring the test step to 7a.

Reply:
Any better suggestion to do this?

Reply:
Added else to skip tests.